### PR TITLE
Split the project into microservices

### PR DESCRIPTION
**Pull request by JunchenLi96**

### CODE REVIEW

File: ExpenseManagerTransactions/DomainModels/DbContext.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using System.Data;
+using System.Data.SqlClient;
+
+namespace ExpenseManagerTransactions.DomainModels
+{
+    public class DbContext
```
Comment (yi-at-rush) [inline ExpenseManagerTransactions/DomainModels/DbContext.cs:6]: This looks the same in both services, and probably never gonna change. Better to extract into a common project.

File: ExpenseManagerTransactions/ExpenseManagerTransactionsDB/ExpenseManagerTransactionsDB.sqlproj [hunk 2]
```diff
@@ -52,7 +52,7 @@
     <SSDTExists Condition="Exists('$(MSBuildExtensionsPath)\Microsoft\VisualStudio\v$(VisualStudioVersion)\SSDT\Microsoft.Data.Tools.Schema.SqlTasks.targets')">True</SSDTExists>
     <VisualStudioVersion Condition="'$(SSDTExists)' == ''">11.0</VisualStudioVersion>
   </PropertyGroup>
-  <Import Condition="'$(NetCoreBuild)' == 'true'" Project="$(NETCoreTargetsPath)\Microsoft.Data.Tools.Schema.SqlTasks.targets"/>
+  <Import Condition="'$(NetCoreBuild)' == 'true'" Project="$(NETCoreTargetsPath)\Microsoft.Data.Tools.Schema.SqlTasks.targets" />
```
Comment (yi-at-rush) [inline ExpenseManagerTransactions/ExpenseManagerTransactionsDB/ExpenseManagerTransactionsDB.sqlproj:55]: It's not common to have one project nested under another project. You could just put all projects under the root, and use solution folders to group projects for each service.

File: ExpenseManagerUsers/Startup.cs [hunk 3]
```diff
@@ -26,6 +26,32 @@ public Startup(IConfiguration configuration)
...
+                return new DbContext(connectionString);
+            });
+
+            services.AddSingleton<IUserService, UserService>();
+
+            services.Configure<JwtConfig>(Configuration.GetSection("TokenOptions"));
+
+            services.AddSingleton<ITokenService, TokenService>();
+
+            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
```
Comment (yi-at-rush) [inline ExpenseManagerUsers/Startup.cs:43]: Is the jwt stuff needed in user service? If so, this looks the same to the transaction service, worth sharing the code. Please also look for other duplicated code between the two services which is stable (not likely to change in the future), try to share them if possible.

File: expense_manager_microservices.sln [hunk 4]
```diff
@@ -7,10 +7,6 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ExpenseManagerUsers", "Expe
 EndProject
```
Comment (yi-at-rush) [inline expense_manager_microservices.sln:7]: Dotnet naming convention is ` ExpenseManagerMicroservices.sln`

File: ExpenseManagerTransactions/Controllers/TransactionsController.cs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+        public TransactionsController(ITransactionService transactionService)
+        {
+            _transactionService = transactionService;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetAllTransactions()
+        {
+            IEnumerable<Transaction> transactions = await _transactionService.GetAllTransactions(ClaimsExtensions.GetUserID(User));
```
Comment (yi-at-rush) [inline ExpenseManagerTransactions/Controllers/TransactionsController.cs:28]: At some point, we need to make this endpoint return some user info in order to show the data sync from user service to transaction service.

**Comment by yi-at-rush** [comment]
And the PR should be from `refactoring/implement-everything` to `master`

File: ExpenseManagerUsers/Startup.cs (hunk 3 above)
Comment (JunchenLi96) [reply ExpenseManagerUsers/Startup.cs:43]: removed JWT bearer and Authentication from Users endpoint as it's not being used

**Pull request merged**